saleor/plugins/webhook/plugin.py

```diff
@@ -114,6 +115,49 @@ def category_deleted(self, category: "Category", previous_value: None) -> None:
                 payload, event_type, webhooks, category, self.requestor
             )
 
+    def channel_created(self, channel: "Channel", previous_value: None) -> None:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.CHANNEL_CREATED
+        if webhooks := get_webhooks_for_event(event_type):
+            payload = {"id": graphene.Node.to_global_id("Channel", channel.id)}
+            trigger_webhooks_async(
+                payload, event_type, webhooks, channel, self.requestor
+            )
+
+    def channel_updated(self, channel: "Channel", previous_value: None) -> None:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.CHANNEL_UPDATED
+        if webhooks := get_webhooks_for_event(event_type):
+            payload = {"id": graphene.Node.to_global_id("Channel", channel.id)}
+            trigger_webhooks_async(
+                payload, event_type, webhooks, channel, self.requestor
+            )
+
+    def channel_deleted(self, channel: "Channel", previous_value: None) -> None:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.CHANNEL_DELETED
+        if webhooks := get_webhooks_for_event(event_type):
+            payload = {"id": graphene.Node.to_global_id("Channel", channel.id)}
+            trigger_webhooks_async(
+                payload, event_type, webhooks, channel, self.requestor
+            )
+
+    def channel_status_changed(self, channel: "Channel", previous_value: None) -> None:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.CHANNEL_STATUS_CHANGED
+        if webhooks := get_webhooks_for_event(event_type):
+            payload = {
+                "id": graphene.Node.to_global_id("Channel", channel.id),
+                "is_active": channel.is_active,
+            }
+            trigger_webhooks_async(
+                payload, event_type, webhooks, channel, self.requestor
+            )
+
```

Comment:
Looking at this module we have a lot of duplications. I think 
```
if not self.active:
    return previous_value
``` 
could be decorated for example `@active_required`

and 
```
if webhooks := get_webhooks_for_event(event_type):
    payload = {"id": graphene.Node.to_global_id("Channel", channel.id)}
    trigger_webhooks_async(
        payload, event_type, webhooks, channel, self.requestor
     )
``` 
would be `trigger_webhooks_for_event(WebhookEventAsyncType.CHANNEL_STATUS_CHANGED)` with building payload etc inside. It's just an idea but for additional statuses there will be more duplications and mess in general. It's out of the scope ofc.

wdyt? @korycins @SzymJ @maarcingebala

Reply:
@L3str4nge  IMO The better way would be to handle this on plugins manager side. And... 🥁 🥁 🥁 
It is already done 😄  (at least for the "default" way of calling plugins)
`saleor.plugins.manager.PluginsManager.__run_method_on_plugins`.

Reply:
So if it's already done it would be awesome to get rid of that mess :D

Reply:
Sounds good, but we should then rework all these methods, not only the new ones from this PR. So maybe for now a maintenance task and we would do it in a separate PR?